seahorn/lib/linked_list_helper.c

```diff
@@ -278,6 +367,72 @@ void aws_linked_list_save_to_head(struct aws_linked_list *list, size_t size,
   }
 }
 
+void aws_linked_list_save_full(struct aws_linked_list *list, size_t size,
+                               struct aws_linked_list_node *start,
+                               struct aws_linked_list_node *end,
+                               struct saved_aws_linked_list *to_save)
+{
+  sea_save_aws_node_to_sea_node(&list->head, &to_save->head);
+  sea_save_aws_node_to_sea_node(&list->tail, &to_save->tail);
+  to_save->save_point = start;
+  to_save->save_point_end = end;
+  if (start == end) {
+    save_one_node(start, to_save);
+    return;
+  }
+  if (size == 0) {
+    /* only save head and tail */
+    if (start == &list->head) {
+      save_two_nodes(start, to_save, getNext);
+    } else {
+      /* save content when empty */
+      to_save->saved_size = 0;
+    }
+  } else if (size == 1) {
+    to_save->saved_size = 0;
+    size_t nodes_idx = 0;
+    /* nodes: [head, head.next...] or [head.next...] */
+    if (start == &list->head) {
+      sea_save_aws_node_to_sea_node(&list->head, &to_save->nodes[nodes_idx]);
```

Comment:
optional: directly store to 0 and 1 instead of nodes_idx to make code readable.

Reply:
absolutely. direct access is either to optimize!

Reply:
should the same apply to tail and tail.prev? their indices are depended on whether head and head.next are stored, so using something similar to `nodes_idx` can still reduce repeated code and make code readable; maybe rename `nodes_idx`->`back_nodes_start`?

Reply:
updated

Reply:
the idea is to use concrete values as much as possible as opposed to symbols because concrete values might be simplified. Performance is as critical as readability here.